blocks/library/gallery/block.js

```diff
@@ -51,33 +52,38 @@ class GalleryBlock extends Component {
 
 		this.state = {
 			selectedImage: null,
+			captionSelected: false,
 		};
 	}
 
 	onSelectImage( index ) {
-		return ( event ) => {
-			// ignore clicks in the editable caption.
-			// Without this logic, text operations like selection, select / unselects the images.
-			if ( event.target.tagName === 'FIGCAPTION' ) {
-				return;
-			}
-
-			this.setState( {
-				selectedImage: index,
+		return () => {
+			this.setState( ( state ) => {
+				// no state change required
+				if ( state.selectedImage === index && ! state.captionSelected ) {
+					return;
```

Comment:
Hm, isn't this equivalent to `this.setState( undefined )`? Shouldn't we avoid these checks inside the `setState` call? Is there much benefit in adding the check at all?

Reply:
Thank you for the feedback @iseulde, In the new approach, checkes are made outside setState, so we are certain no state change occurs if not needed.